### DISCUSSION TITLE
Remote programming of nRF motes

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2833] [Q&A] [answered]
We have a test bed of nRF52840 DK and Dongles. At this time they are connected to Raspberry Pi computers, spread across three floors. When we tried to upload new code to the remote motes, the nRF tools claim that the RPi CPU is not supported, which forces us to collect all motes to a central Intel based computer, each time our code changes.

This exercise may be good for our health but is very laborious and error prone. Is there a way to remotely program the nRF devices?

Will connecting the devices directly to Ethernet switch (vis USB to Ethernet) going to help? Is there any other way that was successful before?

**Comment by pjonsson** [comment, marked as answer]
https://devzone.nordicsemi.com/nordic/nordic-blog/b/blog/posts/nrf-util-unified-command-line-utility has a bunch of comments that request the software to be built for aarch64 too, but no response. There is a comment from the author that links to devzone, so you might be able to find a ticket/thread there where you can comment to indicate you also want that.

In the meantime, you can try the old nrfutil (https://github.com/NordicSemiconductor/pc-nrfutil) that was a Python thing. If you manage to get it to work once, I don't think that will fall apart by itself.
